### DISCUSSION TITLE
Automatically generated Customer Number / ID

### DISCUSSION THREAD

**Discussion by maurice-kpw** [kimai/kimai#3372] [General]
Hello!

Currently the "Account" field can be used to set a custom customer ID / customer number per customer.

I think it would be beneficial to have a setting where a template for future customer numbers can be set. I think the same variables used in the [invoice number format](https://www.kimai.org/documentation/invoices.html#invoice-number-format) would be practicable here. 

For example, we create customer numbers by taking the year and combining it with a counter per year. So we could use "{y}{cy,4}" to automatically set this number.

I also would suggest this number is editable (maybe someone adds a customer to kimai which already has a number) so it would be a suggestion based on the set format. But since every customer number has to be unique there should be a check if the number already exists in kimai.

So the user workflow for creating a new customer would be:

1. Creating a new customer
2. Check if generated number is correct 
3. Change number if necessary
4. Input other relevant data
5. Save the new customer

**Comment by kevinpapst** [comment]
@maurice-kpw Please create the issue here (copy & paste the content): 
https://github.com/kevinpapst/kimai2/issues/new?assignees=&labels=feature+request&template=feature_request.md&title=

Discussion are not followed up. And @BeckeBauer please vote again afterwards 😁

**Reply by maurice-kpw** [reply]
Copied this discussion as an issue: https://github.com/kevinpapst/kimai2/issues/3373

**Reply by kevinpapst** [reply]
Awesome, thanks
